options.go

```diff
@@ -76,7 +76,7 @@ type Options struct {
 	// Default is 3 seconds.
 	ReadTimeout time.Duration
 	// Timeout for socket writes. If reached, commands will fail
-	// with a timeout instead of blocking.
+	// with a timeout instead of blocking. Use value -1 for no timeout and 0 for default.
```

Comment:
`-1 for no timeout` is meant to disable the timeout, but it looks like you always call `SetReadDeadline` no matter what. Perhaps the comment must be updated?